src/identity.rs

```diff
@@ -0,0 +1,844 @@
+    /// Also clippy complains about the `generate_ids` which is called here.
+    /// For this reason, this function is marked as "dead code".
+    ///
+    /// # Arguments
+    /// - `app` : Application name.
+    /// - `instance` : Application name.
+    /// - `eligible_nodes` : Eligible nodes (and pod replicas) grouped by role and group.
+    /// - `id_label_name` : Name of the pod's id label used to store the `id` field of `PodIdentity`
+    /// - `start` : The initial value when generating the `id` fields of pod identities.
+    #[allow(dead_code)]
+    pub fn new(
+        app: &str,
+        instance: &str,
+        eligible_nodes: &EligibleNodesForRoleAndGroup,
+        id_label_name: &str,
+        start: usize,
+    ) -> Self {
+        LabeledPodIdentityFactory {
+            app: app.to_string(),
+            instance: instance.to_string(),
+            id_label_name: id_label_name.to_string(),
+            slice: Self::generate_ids(app, instance, eligible_nodes, start),
+        }
+    }
+
+    /// Returns the given `pod_id` if it's fields match those managed by `Self`
+    /// This is a sanity check to make sure we don't mix pods that were not generated
+    /// using this factory.
+    fn fields_match(&self, pod_id: PodIdentity) -> Result<PodIdentity, Error> {
+        if self.app != pod_id.app() {
+            return Err(Error::UnexpectedPodIdentityField {
+                field: "app".to_string(),
+                value: pod_id.app().to_string(),
+                expected: self.app.clone(),
+            });
+        }
+        if self.instance != pod_id.instance() {
+            return Err(Error::UnexpectedPodIdentityField {
+                field: "instance".to_string(),
+                value: pod_id.instance().to_string(),
+                expected: self.instance.clone(),
+            });
+        }
+        Ok(pod_id)
+    }
+    /// Returns a Vec of pod identities according to the replica per role+group pair from `eligible_nodes`.
+    ///
+    /// The `id` field is in the range from one (1) to the number of replicas per role+group. If no replicas
+    /// are defined, then the range goes from one (1) to the number of eligible groups.
+    ///
+    /// Given a `start` value of 1000, a role with two groups where the first group has two replicas and
+    /// the second has three replicas, the generated `id` fields of the pod identities are counted as follows:
+    ///
+    /// ```yaml
+    /// role_1:
+    ///     - group_1:
+    ///         - id: 1000
+    ///         - id: 1001
+    ///     - group_2:
+    ///         - id: 1002
+    ///         - id: 1003
+    ///         - id: 1004
+    /// ```
+    ///
+    /// # Arguments
+    /// * `app_name` - Application name
+    /// * `instance` - Service instance
+    /// * `eligible_nodes` - Eligible nodes grouped by role and groups.
+    /// * `start` - The starting value for the id field.
+    #[allow(dead_code)] // see comment for Self::new
```

Comment:
You do not need that `#[allow(dead_code)]`. Clippy is not complaining. It is used in new().

Reply:
Done.